cmd/substreams/pack.go

```diff
@@ -20,12 +23,45 @@ var packCmd = &cobra.Command{
 
 func init() {
 	rootCmd.AddCommand(packCmd)
+	packCmd.Flags().StringP("output-dir", "o", ".", cli.FlagDescription(`
+		Optional flag to specify output directory to output generated .spkg file. 
+		If the received argument starts with a "." it will be treated as a local path relative to the supplied manifest path.
+	`))
+}
+
+func is_relative_path(path string) bool {
+	return path[0:1] == "."
 }
 
 func runPack(cmd *cobra.Command, args []string) error {
+	outputDir := maybeGetString(cmd, "output-dir")
+	if outputDir == "." {
+		outputDir = ""
+	}
+
 	manifestPath := args[0]
 
+	if outputDir != "" {
+		if is_relative_path(outputDir) {
+			workingDirectory, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("can't retrieve current directory information")
+			}
+			newOutputDir := filepath.Join(filepath.Dir(filepath.Join(workingDirectory, manifestPath)), outputDir)
+			fmt.Printf("Output directory specified: %s \nThis will be treated as a local path.\nFull folderpath: %s\n", outputDir, newOutputDir)
+			outputDir = newOutputDir
+		} else {
+			fmt.Printf("Output directory treated as an absolute path.\nFull folderpath: %s\n", outputDir)
+		}
+
+		err := os.MkdirAll(outputDir, os.ModePerm)
+		if err != nil {
+			return fmt.Errorf("error creating output directory: %w", err)
+		}
+	}
```

Comment:
All this is not needed, `os.MkdirAll` and `ioutil.WriteFile` (and any other I/O path related functions) automatically make relative path absolute against the process working directory, so it's not needed at all to make the path absolute.
```suggestion
        if err := os.MkdirAll(outputDir, os.ModePerm); err != nil {
	        return fmt.Errorf("creating output directories: %w", err)
        }
```